httpclient5/src/test/java/org/apache/hc/client5/http/SystemDefaultDnsResolverTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+package org.apache.hc.client5.http;
+
+import org.junit.jupiter.api.Test;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SystemDefaultDnsResolverTest {
+
+    @Test
+    void resolve() throws UnknownHostException {
+        final SystemDefaultDnsResolver resolver = SystemDefaultDnsResolver.INSTANCE;
+
+        final InetAddress[] result1 = resolver.resolve("example.com");
```

Comment:
@arturobernalg While extra test coverage is nice, in this particular case these tests introduce dependency on DNS infrastructure with public network access. These tests may fail when run in a private network where 'example.com' does not resolve. Please re-write the tests that do not require public network access or remote them.